### ISSUE TITLE
Inconsistent formatting before and after braces in source files

### ISSUE THREAD

**Issue by Anutrix** [godotengine/godot#29936]
**Godot version:**
Godot master branch https://github.com/godotengine/godot/commit/d6f8a43b600cc5236c5860c8f266b6f8afc9c23c

**Issue description:**
**What happened**:
Inconsistency 1. Source files sometimes do and sometimes don't seem to have newlines after each function's ending brace.(Marked blue in screenshot).
Inconsistency 2. Source files sometimes do and sometimes don't seem to have newlines after each function's starting brace.(Marked red in screenshot).
**What should happen**:
Consistent styles regarding above.

**Steps to reproduce:**
Search sources files. 
e.g, Lines after [texture.cpp:1155](https://github.com/godotengine/godot/blob/7b88ac08437d3bd062efcbdd40c215b533032412/scene/resources/texture.cpp#L1155) show an example the 2 formatting inconsistencies.

**Minimal reproduction project:**

![Capture45](https://user-images.githubusercontent.com/11459028/59869593-49503b00-93b1-11e9-9336-dc432eee2aae.PNG)

Can clang-format do anything about it?

**Comment by Xrayez** [comment]
1. I'd make this consistent by having space between functions indeed.
2. Having an empty space in function body allows to add more code easily. Also if there's no space, the code bleeds into the function signature making it somewhat difficult to comprehend, but that might also depend on the line spacing defined by font used.

**Comment by akien-mga** [comment]
1. is a habit of @reduz to group setters and getters together this way. It's not used consistently and does not provide so much value, so I'd be fine enforcing a line between all methods if there's a way to do so. I don't recall a clang-format option for that sadly.

2. also comes from @reduz's original coding style which most of us got used to and use ourselves, though it's also not used consistently and hard to enforce one way or another. The space makes the function signature stand out and makes it clear where the body of the function starts. One could also use the opening bracket on that line for a similar effect, but that's a whole can of worms I'm not willing to open (and I'm not in favor of changing our brace style anyway).

**Comment by Anutrix** [comment]
Is it Ok if I do it manually or using some scripts and make pull request(s)?
It might be better to do this over many pull requests as changes might be large. And doing it over many updates might let us do it more safely.

**Comment by akien-mga** [comment]
Not until we reach a consensus on whether a given style should be enforced.

**Comment by YeldhamDev** [comment]
Related: #16521

**Comment by akien-mga** [comment]
Superseded by #33027.
